### PR TITLE
Dev feat509 hashing v0.2

### PR DESCRIPTION
**Pull request by serializingme**
New version of #1276. Created a single file for filemd5, filesha1 and filesha256 keywords. If it doesn't work for you lads, I can always change to four different files  (one with common code, and the rest with the specific code of the hashing algorithms).

### CODE REVIEW

File: src/util-file.c [hunk 1]
```diff
@@ -493,13 +529,25 @@ int FileAppendData(FileContainer *ffc, const uint8_t *data, uint32_t data_len)
 
     if (FileStoreNoStoreCheck(ffc->tail) == 1) {
 #ifdef HAVE_NSS
-        /* no storage but forced md5 */
+        /* no storage but forced md5, sha1 or/and sha256 */
         if (ffc->tail->md5_ctx) {
             if (ffc->tail->md5_ctx)
```
Comment (serializingme) [inline src/util-file.c]: I have replicated this bellow, but I'm unsure why it has two if statements with the same condition. Shouldn't this be only one if?

Comment (inliniac) [inline src/util-file.c]: Looks like it ya.

Comment (serializingme) [inline src/util-file.c]: Ok, fix will go in the next version of the pull request.

**Comment by inliniac** [comment]
Replaced by #2025

**Pull request closed without merging**